skimage/segmentation/_join.py

```diff
@@ -132,7 +132,9 @@ def relabel_sequential(label_field, offset=1):
     if np.dtype(required_type).itemsize > np.dtype(label_field.dtype).itemsize:
         label_field = label_field.astype(required_type)
     new_labels0 = np.arange(offset, offset + len(labels0))
-    if np.all(labels0 == new_labels0):
+    if offset == 1 and np.all(labels0 == new_labels0):
+        if not (labels == 0).any():
+            labels = np.concatenate(([0], labels))
         return label_field, labels, labels
```

Comment:
Maybe it would be best to not handle this special case separately and simply remove these lines of code.

Reply:
@uschmidt83 I'm actually confused about what's happening in this edge case. The test example has a lot going on so it's still not clear to me. Could you elaborate on the problem you're fixing and why it goes wrong in the current code?

Reply:
Hi @jni, these lines intended to check if `label_field` already contains sequential labels, hence there would be nothing to do. Unfortunately, the logic only works if `offset == 1` and `label_field` contains the 0 label. Hence, I added these things.

Reply:
This check was probably intended to avoid some computation, but it is unnecessary.
I just pushed a commit that (IMO) simplifies the function and makes it easier to understand and maintain.